### PR TITLE
settings this.servers when its undefined

### PR DESCRIPTION
**Pull request by doron2402**
when this.servers is undefined we can just return a random server, right now it will return a number as a server.

is there a list of default server urls?

@andyroyle

### CODE REVIEW

**Comment by andyroyle** [comment]
we can't risk hard-coding servers into the client because the discovery servers run on mesos, so are liable to change regularly. In the event that this.servers is empty then it'll return undefined, which will cause an error later on in the process. I think this is fine for now.

File: discovery.js [hunk 1]
```diff
@@ -26,6 +26,9 @@ DiscoveryClient.prototype._unbackoff = function() {
 };
 
 DiscoveryClient.prototype._randomServer = function() {
+  if (!this.servers || !this.servers.length) {
+    return new Error("There no servers defined.");
```
Comment (doron2402) [inline discovery.js]: @andyroyle what about now?

Comment (doron2402) [inline discovery.js]: @stevenschlansker ^^

**Comment by stevenschlansker** [comment]
Would it make more sense to catch this in the `connect()` and `reconnect()` functions?  Bail out early if we can't find any servers?  Also, it seems that returning an `Error` doesn't make any sense because the callers do not check for it.  They do a string append.  i.e. `var url = _randomServer() + "/watch";` which seems like it will work really poorly.

**Comment by doron2402** [comment]
I agree, but we need to fix this issue when server are not defined

**Comment by stevenschlansker** [comment]
Happy to merge a PR that improves behavior, but I don't think this change fixes anything, I think it just introduces an even more bizarre failure mode.  Instead of doing `undef + "string"` you do `new Error() + "string"`.

**Comment by andyroyle** [comment]
I agree with @stevenschlansker, calling console.error is even worse.

If the servers list is empty during a connect, then we can treat that as an unrecoverable error and bail out. If the servers list becomes empty during runtime then the client can still continue to serve leases from it's stale cache.

**Pull request merged**